### PR TITLE
Create SymbolToken and incorporate to SymbolTable

### PR DESCRIPTION
**Pull request by byronlin-bq**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ion/symboltable.go [hunk 1]
```diff
@@ -28,7 +28,8 @@ type SymbolTable interface {
 	Symbols() []string
 	// MaxID returns the maximum ID this symbol table defines.
 	MaxID() uint64
-
+	// Find finds the the SymbolToken by its name.
```
Comment (therapon) [inline ion/symboltable.go:31]:
```suggestion
	// Find finds the  SymbolToken by its name.
```

File: ion/symboltoken.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	return &value
+}
+
+// Equal figures out if two import sources are equal for each component.
+func (is *ImportSource) Equal(o *ImportSource) bool {
+	return is.Table == o.Table && is.SID == o.SID
+}
+
+// SymbolToken are the values that annotations, field names, and the textual content of Ion symbol values.
+// The `nil` value for SymbolToken is $0.
```
Comment (therapon) [inline ion/symboltoken.go:50]:
```suggestion
// The `nil` value for SymbolToken is the SID `$0`.
```

File: ion/symboltoken.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	}
+	return &value
+}
+
+// Equal figures out if two import sources are equal for each component.
+func (is *ImportSource) Equal(o *ImportSource) bool {
+	return is.Table == o.Table && is.SID == o.SID
+}
+
+// SymbolToken are the values that annotations, field names, and the textual content of Ion symbol values.
```
Comment (therapon) [inline ion/symboltoken.go:49]:
```suggestion
// SymbolToken is the representation fir annotations, field names, and the textual content of Ion symbol values.
```

File: ion/symboltoken.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	Text *string
+	// Local symbol ID associated with the token.
+	LocalSID int64
+	// The shared symbol table location that this token came from, or nil if undefined.
+	Source *ImportSource
+}
+
+var (
+	// symbolTokenUndefined is the sentinel for invalid tokens.
+	// The `nil` value is actually $0 which is a defined token.
```
Comment (therapon) [inline ion/symboltoken.go:62]:
```suggestion
	// The `nil` value is actually SID `$0` which is a defined token.
```

File: ion/symboltoken_test.go [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+}{
+	{nil, 123, nil,
+		nil, 123, nil},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table", 1)},
+	{newString("text1"), 123, nil,
+		newString("text1"), 123, nil},
+	{newString("text2"), 123, newSource("table", 1),
+		newString("text2"), 123, newSource("table", 1)},
+}
```
Comment (therapon) [inline ion/symboltoken_test.go:45]: Indentation looks off to me (maybe its the PR view in the browser?)
Even though LocalSID is not taken into account for equality, the test should have test cases were the value differs

File: ion/symboltoken_test.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+		nil, 123, nil},
+	{nil, 123, nil,
+		newString("text1"), 123, nil},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table2", 1)},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table", 2)},
+	{newString("text2"), 123, nil,
+		newString("text3"), 123, nil},
+}
```
Comment (therapon) [inline ion/symboltoken_test.go:76]: similar comment here about indentation and test cases

File: ion/symboltoken_test.go [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+		t.Errorf("expected %v, got %v", 1, is.SID)
+	}
+}
+
+var ImportSourceBoolEqualsTestData = []struct {
+	text1 string
+	sid1  int64
+	text2 string
+	sid2  int64
+}{
```
Comment (therapon) [inline ion/symboltoken_test.go:151]: If we add an extra element here for the expected value of test (true or false) then the two sets of tests can be collapsed to 1 set of tests.

File: ion/symboltable.go (hunk 1 above)
Comment (byronlin-bq) [reply ion/symboltable.go]: Updated.

File: ion/symboltoken.go (hunk 2 above)
Comment (byronlin-bq) [reply ion/symboltoken.go]: Updated.

File: ion/symboltoken.go (hunk 3 above)
Comment (byronlin-bq) [reply ion/symboltoken.go]: Updated.

File: ion/symboltoken.go (hunk 4 above)
Comment (byronlin-bq) [reply ion/symboltoken.go]: Updated.

File: ion/symboltoken_test.go (hunk 5 above)
Comment (byronlin-bq) [reply ion/symboltoken_test.go]: Updated.

File: ion/symboltoken_test.go (hunk 6 above)
Comment (byronlin-bq) [reply ion/symboltoken_test.go]: Updated.

File: ion/symboltoken_test.go (hunk 7 above)
Comment (byronlin-bq) [reply ion/symboltoken_test.go]: Updated.

**Pull request merged**